### PR TITLE
add test steps, refresh uploaded manifest file

### PR DESCRIPTION
**Pull request by vijaysawant**
### Problem Statement
As per PX recommendation need to add steps and update automation coverage around the test.
Jira-23358

### Solution
- Add test steps in test case
- Add additional assersions at the end of test

### Related Issues
N/A

### CODE REVIEW

**Comment by vijaysawant** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_repository.py -k test_missing_content_id

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 5854
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_repository.py -k test_missing_content_id
Test Result : ========== 1 passed, 165 deselected, 11 warnings in 735.28s (0:12:15) ==========
```

**Comment by vsedmik** [comment]
CI/CQ seems unrelated to this PR, rather related to #13211 @pondrejk 
```
test_positive_eol_banner_e2e:38
-------------------------------

* Unexpected tokens:
  Caselevel: Acceptance
  Testtype: Functional
  Upstream: No
```

**Review by ogajduse — APPROVED** [review]
ACK, but I, as a reviewer, would like to have more context about the change and what PX requests. That could be achieved by referencing a Jira issue in the PR description.

One minor comment left.

File: tests/foreman/api/test_repository.py [hunk 1]
```diff
@@ -1521,14 +1521,22 @@ def test_positive_sync_kickstart_check_os(
         ),
         indirect=True,
     )
-    def test_missing_content_id(self, repo):
+    def test_missing_content_id(self, repo, module_sca_manifest_org, module_target_sat):
```
Comment (shweta83) [inline tests/foreman/api/test_repository.py:1524]: We are still using sca-disabled manifest in 6.14.z and lower versions. will you please cherrypick it to 6.15.z only and have another PR for 6.14.z and lower?

**Review by shweta83 — CHANGES_REQUESTED** [review]

File: tests/foreman/api/test_repository.py (hunk 1 above)
Comment (vijaysawant) [reply tests/foreman/api/test_repository.py:1524]: Ok I will discuss with Phoenix team and will take appropriate action.

File: tests/foreman/api/test_repository.py [hunk 2]
```diff
@@ -1546,6 +1554,10 @@ def test_missing_content_id(self, repo):
         repo.delete()
         with pytest.raises(HTTPError):
             repo.read()
+        output = module_target_sat.cli.Subscription.refresh_manifest(
+            {'organization-id': module_sca_manifest_org.id}
+        )
+        assert "Candlepin job status: SUCCESS" in output, "Failed to refresh manifest"
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_repository.py:1560]:
```suggestion
        assert 'Candlepin job status: SUCCESS' in output, 'Failed to refresh manifest'
```

File: tests/foreman/ui/test_eol_banner.py [hunk 3]
```diff
@@ -4,17 +4,12 @@
 
 :CaseAutomation: Automated
 
-:CaseLevel: Acceptance
-
 :CaseComponent: Dashboard
 
 :Team: Endeavour
 
-:TestType: Functional
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_eol_banner.py:13]: @vijaysawant Could you keep a separate commit for fixing this please?

File: tests/foreman/api/test_repository.py [hunk 4]
```diff
@@ -1521,14 +1521,22 @@ def test_positive_sync_kickstart_check_os(
...
-        :expectedresults: Repository URL can be set to something new and the repo can be deleted
+        :setup:
+            1. Create product and repo, sync repo
+
+        :steps:
+            1. Try to update repo URL
+            2. Attempt to delete repo
+            3. Refresh manifest file
+
+        :expectedresults: Repository URL can be set to something new, repo can be deleted and manifest was refreshed successfully
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_repository.py:1539]:
```suggestion
        :expectedresults: Repo URL can be updated, repo can be deleted and manifest refresh works after repo delete
```

File: tests/foreman/ui/test_eol_banner.py (hunk 3 above)
Comment (Gauravtalreja1) [reply tests/foreman/ui/test_eol_banner.py:13]: OR open a separate PR instead as this eol banner tests added by @pondrejk are only applicable for master and 6.15.z, and for your PR to cherrypick to below versions auto-cherrypick workflow will fail because of this :)

**Comment by vijaysawant** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_repository.py -k test_missing_content_id

**Comment by vijaysawant** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_repository.py -k test_missing_content_id

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 5880
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_repository.py -k test_missing_content_id --external-logging
Test Result : ========== 1 passed, 165 deselected, 11 warnings in 708.47s (0:11:48) ==========
```

File: tests/foreman/api/test_repository.py (hunk 1 above)
Comment (vijaysawant) [reply tests/foreman/api/test_repository.py]: new PR has been opened please take a look
https://github.com/SatelliteQE/robottelo/pull/14208

**Review by vsedmik — APPROVED** [review]
